iceoryx_binding_c/test/moduletests/test_server.cpp

```diff
@@ -0,0 +1,428 @@
+{
+    ::testing::Test::RecordProperty("TEST_ID", "10091749-4c07-442a-b42b-aef0d8d06ceb");
+    ServerOptions defaultOptions;
+    prepareServerInit(defaultOptions);
+
+    iox_server_t sut = iox_server_init(&sutStorage, SERVICE, INSTANCE, EVENT, nullptr);
+    ASSERT_THAT(sut, Ne(nullptr));
+}
+
+TEST_F(iox_server_test, InitializingServerWithCustomOptionsWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "e21f9257-cf3b-4e18-ad98-126a295e780f");
+    iox_server_options_t options;
+    iox_server_options_init(&options);
+    options.requestQueueCapacity = 32;
+    strncpy(options.nodeName, "do not hassel with the hoff", IOX_CONFIG_NODE_NAME_SIZE);
+    options.offerOnCreate = false;
+    options.requestQueueFullPolicy = QueueFullPolicy_BLOCK_PRODUCER;
+    options.clientTooSlowPolicy = ConsumerTooSlowPolicy_WAIT_FOR_CONSUMER;
+
+    ServerOptions cppOptions;
+    cppOptions.requestQueueCapacity = 32;
+    cppOptions.nodeName = "do not hassel with the hoff";
+    cppOptions.offerOnCreate = false;
+    cppOptions.requestQueueFullPolicy = iox::popo::QueueFullPolicy::BLOCK_PRODUCER;
+    cppOptions.clientTooSlowPolicy = iox::popo::ConsumerTooSlowPolicy::WAIT_FOR_CONSUMER;
+
+    prepareServerInit(cppOptions);
+
+    iox_server_t sut = iox_server_init(&sutStorage, SERVICE, INSTANCE, EVENT, &options);
+    ASSERT_THAT(sut, Ne(nullptr));
+}
+
+TEST_F(iox_server_test, DeinitReleasesServer)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "a50eb082-bc40-4c84-95dc-cefcdfcdb70d");
+    prepareServerInit();
+    iox_server_t sut = iox_server_init(&sutStorage, SERVICE, INSTANCE, EVENT, nullptr);
+    iox_server_deinit(sut);
+    EXPECT_THAT(sutPort->m_toBeDestroyed.load(), Eq(true));
+}
+
+TEST_F(iox_server_test, WhenNotOfferedTakeRequestFails)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "48763e8a-c192-4baf-88c0-fb5377d89127");
+    prepareServerInit();
+    iox_server_t sut = iox_server_init(&sutStorage, SERVICE, INSTANCE, EVENT, nullptr);
+    iox_server_stop_offer(sut);
+
+    const void* payload;
+    EXPECT_THAT(iox_server_take_request(sut, &payload),
+                Eq(ServerRequestResult_NO_PENDING_REQUESTS_AND_SERVER_DOES_NOT_OFFER));
+}
+
+TEST_F(iox_server_test, WhenOfferedAndNoRequestsPresentTakeFails)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "409ab402-2452-4250-a6b4-4c90489cdd42");
+    prepareServerInit();
+    iox_server_t sut = iox_server_init(&sutStorage, SERVICE, INSTANCE, EVENT, nullptr);
+    iox_server_offer(sut);
+
+    const void* payload;
+    EXPECT_THAT(iox_server_take_request(sut, &payload), Eq(ServerRequestResult_NO_PENDING_REQUESTS));
+}
+
+TEST_F(iox_server_test, WhenOfferedAndRequestsPresentTakeSucceeds)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "42443d18-5a8f-40ee-938d-2b0a345f829b");
+    prepareServerInit();
+    iox_server_t sut = iox_server_init(&sutStorage, SERVICE, INSTANCE, EVENT, nullptr);
+    iox_server_offer(sut);
+    receiveRequest(64461001);
```

Comment:
Magic number. It seems the parameter for `receiveRequest` is only used in one tests. Could make the parameter optional with a default value and remove the magic numbers from the tests where they aren't checked